OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -93,13 +93,16 @@ void IRulesetLoaded<WeaponInfo>.RulesetLoaded(Ruleset rules, WeaponInfo wi)
 		}
 	}
 
-	public class AreaBeam : IProjectile, ISync
+	public class AreaBeam : IProjectile, ISync, ISpatiallyPartitionable
 	{
 		readonly AreaBeamInfo info;
 		readonly ProjectileArgs args;
 		readonly AttackBase actorAttackBase;
 		readonly Color color;
 		readonly WDist speed;
+		Size screenMapBounds;
+		WPos beamCenter;
+		WVec beamDelta;
```

Comment:
Do these need to be fields? It looks like you don't actually need to maintain them beyond passing them directly to the screen map - would they be fine as locals?